### PR TITLE
More cleanups, more events

### PR DESCRIPTION
**Pull request by revolxof**
Edit + delete need to be implemented at frontend

### CODE REVIEW

**Comment by revolxof** [comment]
I don't know why GitHub didn't register some commits in the previous pr, it's annoying and stupid

File: client/src/connection.ts [hunk 1]
```diff
@@ -94,6 +94,16 @@ export function connect() {
     store.dispatch("channels/addMessage", message)
   });
 
+  // eslint-disable-next-line
+  connection.on("message:edit", async (editMessage: ServerEditMessage) => {
```
Comment (velddev) [inline client/src/connection.ts]: message:update is recommended to follow CRUD conventions

File: client/src/models/index.ts [hunk 2]
```diff
@@ -18,12 +18,25 @@ export type ServerMessage = {
   user: string;
   embed?: Embed;
   content: string;
-  mentions: string[];
+  sentTimestamp: number;
+  mentions: Array<string>;
+};
+
+export type ServerEditMessage = {
```
Comment (velddev) [inline client/src/models/index.ts]: is it possible to extend message create from message edit?

File: client/src/models/index.ts [hunk 3]
```diff
@@ -39,7 +52,7 @@ export type Channel = {
   id: string;
   system: boolean;
   name: string;
-  members: string[];
+  members: Array<string>;
```
Comment (velddev) [inline client/src/models/index.ts]: Not really needed?

**Review by velddev — CHANGES_REQUESTED** [review]

File: client/src/connection.ts (hunk 1 above)
Comment (revolxof) [reply client/src/connection.ts]: done

File: client/src/models/index.ts (hunk 2 above)
Comment (revolxof) [reply client/src/models/index.ts]: sure, ill have a look

**Review by velddev — APPROVED** [review]

**Pull request merged**